qiita_db/analysis.py

```diff
@@ -883,47 +895,44 @@ def _build_mapping_file(self, samples):
                      WHERE filepath_type = 'qiime_map'
                         AND artifact_id IN (SELECT *
                                             FROM qiita.find_artifact_roots(%s))
-                     ORDER BY filepath_id DESC"""
-            _id, fp = qdb.util.get_mountpoint('templates')[0]
+                     ORDER BY filepath_id DESC LIMIT 1"""
+            _, fp = qdb.util.get_mountpoint('templates')[0]
+
+            all_ids = set()
             to_concat = []
+            for aid, samps in viewitems(samples):
+                qdb.sql_connection.TRN.add(sql, [aid])
+                qm_fp = qdb.sql_connection.TRN.execute_fetchindex()[0][1]
```

Comment:
Just realized - if you are accession only to `[0][1]`, doesn't it mean that the above query is retrieving more info than needed? I think it should only be `SELECT filepath` and then here you can use `execute_fetchlast`. I think it will make the code cleaner and also removes the magic numbers.